webSockets/server/server_test.go

```diff
@@ -0,0 +1,105 @@
+package server
+
+import (
+	"errors"
+	"sync"
+	"sync/atomic"
+	"testing"
+
+	"github.com/multiversx/mx-chain-core-go/core/mock"
+	"github.com/multiversx/mx-chain-core-go/data/typeConverters/uint64ByteSlice"
+	"github.com/multiversx/mx-chain-core-go/testscommon"
+	"github.com/multiversx/mx-chain-core-go/webSockets"
+	"github.com/multiversx/mx-chain-core-go/webSockets/data"
+	"github.com/stretchr/testify/require"
+)
+
+func createArgs() ArgsWebSocketsServer {
+	payloadConverter, _ := webSockets.NewWebSocketPayloadConverter(uint64ByteSlice.NewBigEndianConverter())
+	return ArgsWebSocketsServer{
+		RetryDurationInSeconds: 1,
+		BlockingAckOnError:     false,
+		WithAcknowledge:        false,
+		URL:                    "url",
+		PayloadConverter:       payloadConverter,
+		Log:                    &mock.LoggerMock{},
+	}
+}
+
+func TestNewWebSocketsServer(t *testing.T) {
+	t.Parallel()
+
+	t.Run("should work", func(t *testing.T) {
+		args := createArgs()
+		ws, err := NewWebSocketsServer(args)
+		require.NotNil(t, ws)
+		require.Nil(t, err)
+		require.False(t, ws.IsInterfaceNil())
+	})
+
+	t.Run("empty url, should return error", func(t *testing.T) {
+		args := createArgs()
+		args.URL = ""
+		ws, err := NewWebSocketsServer(args)
+		require.Nil(t, ws)
+		require.Equal(t, err, data.ErrEmptyUrl)
```

Comment:
require: t, expected, actual
for the entire file

Reply:
done